elastic/kibana
# [ResponseOps][Alerts] Add alerts grouping aggregations endpoint

## Summary

Adds an endpoint dedicated to fetching alerts group aggregations to avoid adding runtime mappings and client-side controlled scripts to the `internal/rac/alerts/find` endpoint.
The new endpoint injects a `groupByField` runtime field used to normalize the values of the field used for grouping, to account for null and multi-element arrays.

#184635 depends on this

Closes #186383

## To verify

Review the added [tests](x-pack/plugins/rule_registry/server/routes/get_alerts_group_aggregations.test.ts).
Use the Kibana Dev Console to test various body params and aggregations:

1. Create any type of rule that fire alerts
2. Wait for the alerts to be created
3. Call the `_group_aggregations` endpoint, using the feature id(s) that cover the type of rules you used:
```
POST kbn:internal/rac/alerts/_group_aggregations
{
   "featureIds": [...],
  ...
}
```
See [here](https://github.com/elastic/kibana/pull/186475/files#diff-0780f60b57fdaa96eda1ab2853064033477617430a17cdb87750cef42c6e8668R22) and [here](https://github.com/elastic/kibana/pull/186475/files#diff-0780f60b57fdaa96eda1ab2853064033477617430a17cdb87750cef42c6e8668R37) to know the available params and pre-defined aggregations.

### Checklist
